From b316f451fb7b001dff828c65224b0946b5572bc7 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 21 May 2025 14:07:00 +0300
Subject: [PATCH] [JSW-704] jgs: use new macro from jgs specs

specs has updated with new naming, this
patch will adapt those changes.

Change-Id: Iab8f8e715ce582b59b05a9ce35534b752aa3a440
---

diff --git a/arch/jgs/common.c b/arch/jgs/common.c
index 3c4afa7..64b8f08 100644
--- a/arch/jgs/common.c
+++ b/arch/jgs/common.c
@@ -17,6 +17,7 @@
 #include <jgs_efuse.h>
 #include <arc_regs.h>
 #include <arc_masks.h>
+#include <soc.h>
 
 #ifdef __ZEPHYR__
 #include <preboot.h>
@@ -94,10 +95,10 @@
 u64 get_64bit_addr_from_32bit_addr(uintptr_t base)
 {
 	/* ARC.MNG DCCM and ICCM regions */
-	if (base >= LOCAL_ICCM_BASE && base < IO_MNG_ICCM_SIZE)
+	if (base >= LOCAL_ICCM_BASE && base < IO_0_MNG_ARC_ICCM_SIZE)
 		return (base & ~IO0_MNG_ICCM_BASE) + IO0_MNG_ICCM_FULL_BASE;
 
-	if (base >= LOCAL_DCCM_BASE && base < IO_MNG_DCCM_SIZE)
+	if (base >= LOCAL_DCCM_BASE && base < IO_0_MNG_ARC_DCCM_SIZE)
 		return (base & ~IO0_MNG_DCCM_BASE) + IO0_MNG_DCCM_FULL_BASE;
 
 	/* TODO Other regions better be added as well (GT, HBW, DBG) */
@@ -115,10 +116,12 @@
 u32 fs1_get_arc_addr_from_64bit(u64 addr)
 {
 	/* ARC.MNG DCCM and ICCM regions */
-	if (addr >= IO0_MNG_ICCM_FULL_BASE && addr < IO0_MNG_ICCM_FULL_BASE + (u64)IO_MNG_ICCM_SIZE)
+	if (addr >= IO0_MNG_ICCM_FULL_BASE && addr < IO0_MNG_ICCM_FULL_BASE +
+		(u64)IO_0_MNG_ARC_ICCM_SIZE)
 		return (u32)((addr & ADDR_MASK(IO_MNG_ICCM)) + LOCAL_ICCM_BASE);
 
-	if (addr >= IO0_MNG_DCCM_FULL_BASE && addr < IO0_MNG_DCCM_FULL_BASE + (u64)IO_MNG_DCCM_SIZE)
+	if (addr >= IO0_MNG_DCCM_FULL_BASE && addr < IO0_MNG_DCCM_FULL_BASE +
+		(u64)IO_0_MNG_ARC_DCCM_SIZE)
 		return (u32)((addr & ADDR_MASK(IO_MNG_DCCM)) + LOCAL_DCCM_BASE);
 
 	/* TODO Other regions better be added as well (GT, HBW, DBG) */
